include/adaflo/sharp_interface_util.h

```diff
@@ -0,0 +1,1108 @@
+                                                             n_points);
+
+      for (unsigned int q = 0; q < n_points; ++q)
+        phi_normal_force.submit_value(JxW[q], q);
+
+      phi_normal_force.integrate(cell, unit_points, buffer, EvaluationFlags::values);
+
+      constraints.distribute_local_to_global(buffer, local_dof_indices, force_vector);
+    }
+}
+
+
+
+template <int dim, int spacedim, typename VectorType>
+void
+compute_normal(const Mapping<dim, spacedim> &   mapping,
+               const DoFHandler<dim, spacedim> &dof_handler_dim,
+               VectorType &                     normal_vector)
+{
+  FEValues<dim, spacedim> fe_eval_dim(mapping,
+                                      dof_handler_dim.get_fe(),
+                                      dof_handler_dim.get_fe().get_unit_support_points(),
+                                      update_normal_vectors | update_gradients);
+
+  Vector<double> normal_temp;
+
+  for (const auto &cell : dof_handler_dim.get_triangulation().active_cell_iterators())
+    {
+      TriaIterator<DoFCellAccessor<dim, spacedim, false>> dof_cell_dim(
+        &dof_handler_dim.get_triangulation(),
+        cell->level(),
+        cell->index(),
+        &dof_handler_dim);
+
+      fe_eval_dim.reinit(dof_cell_dim);
+
+      normal_temp.reinit(fe_eval_dim.dofs_per_cell);
+      normal_temp = 0.0;
+
+      for (const auto q : fe_eval_dim.quadrature_point_indices())
+        {
+          const auto normal = fe_eval_dim.normal_vector(q);
+
+          const unsigned int comp =
+            dof_handler_dim.get_fe().system_to_component_index(q).first;
+
+          normal_temp[q] = normal[comp];
+        }
+
+      dof_cell_dim->set_dof_values(normal_temp, normal_vector);
+    }
+}
+
+
+
+template <int dim, int spacedim, typename VectorType>
+void
+compute_curvature(const Mapping<dim, spacedim> &   mapping,
+                  const DoFHandler<dim, spacedim> &dof_handler_dim,
+                  const DoFHandler<dim, spacedim> &dof_handler,
+                  const Quadrature<dim>            quadrature,
+                  const VectorType &               normal_vector,
+                  VectorType &                     curvature_vector)
+{
+  FEValues<dim, spacedim> fe_eval(mapping,
+                                  dof_handler.get_fe(),
+                                  quadrature,
+                                  update_gradients);
+  FEValues<dim, spacedim> fe_eval_dim(mapping,
+                                      dof_handler_dim.get_fe(),
+                                      quadrature,
+                                      update_gradients);
+
+  Vector<double> curvature_temp;
+
+  for (const auto &cell : dof_handler.get_triangulation().active_cell_iterators())
+    {
+      TriaIterator<DoFCellAccessor<dim, spacedim, false>> dof_cell(
+        &dof_handler.get_triangulation(), cell->level(), cell->index(), &dof_handler);
```

Comment:
I think this is not needed because that is the `dof_handler` over which you iterate with the for loop, as opposed to `dof_cell_dim` below.